src/lib.rs

```diff
@@ -1,14 +1,178 @@
-//! Linked lists that supports arbitrary removal in constant time.
-//!
-//! It is based on the linked list implementation in [Rust-for-Linux][1].
-//!
-//! [1]: https://github.com/Rust-for-Linux/linux/blob/rust/rust/kernel/linked_list.rs
-
-#![no_std]
+#![cfg_attr(not(test), no_std)]
+#![doc = include_str!("../README.md")]
 
 mod linked_list;
+mod raw_list;
+pub use linked_list::List;
+pub use raw_list::{GetLinks, Links};
+
+#[macro_export(local_inner_macros)]
+#[doc(hidden)]
+macro_rules! __def_node_internal {
+    ($(#[$meta:meta])* $vis:vis struct $name:ident($type:ty);) => {
+        $(#[$meta])*
+        $vis struct $name {
+            inner: $type,
+            links: $crate::Links<Self>,
+        }
+
+        impl $crate::GetLinks for $name {
+            type EntryType = Self;
+
+            #[inline]
+            fn get_links(t: &Self) -> &$crate::Links<Self> {
+                &t.links
+            }
+        }
+
+        impl $name {
+            #[doc = "Create a node"]
+            $vis const fn new(inner: $type) -> Self {
+                Self {
+                    inner,
+                    links: $crate::Links::new(),
+                }
+            }
+
+            #[inline]
+            #[doc = "Return the referece of wrapped inner"]
+            $vis const fn inner(&self) -> &$type {
+                &self.inner
+            }
+
+            #[inline]
+            #[doc = "Consumes the `node`, returning the wrapped inner"]
+            $vis fn into_inner(self) -> $type {
```

Comment:
const fn

Reply:
so, in generic type it would complain ` error[E0493]: destructor of `GenericNode<T>` cannot be evaluated at compile-time`  

add for normal node